core/sdk-core/src/main/java/software/amazon/awssdk/core/FileTransformerConfiguration.java

```diff
@@ -0,0 +1,155 @@
+import software.amazon.awssdk.utils.builder.CopyableBuilder;
+import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
+
+/**
+ * Configuration options for {@link AsyncResponseTransformer#toFile(Path, FileTransformerConfiguration)}. All values are optional.
+ */
+@SdkPublicApi
+public final class FileTransformerConfiguration implements ToCopyableBuilder<FileTransformerConfiguration.Builder,
+    FileTransformerConfiguration> {
+    private final FileWriteOption fileModificationOption;
+    private final Boolean deleteOnFailure;
+
+    private FileTransformerConfiguration(DefaultBuilder builder) {
+        this.fileModificationOption = builder.fileWriteOption;
+        this.deleteOnFailure = builder.deleteOnFailure;
+    }
+
+    /**
+     * The configured {@link FileWriteOption} or {@link Optional#empty()} if not configured
+     */
+    public Optional<FileWriteOption> fileModificationOption() {
+        return Optional.ofNullable(fileModificationOption);
+    }
+
+    /**
+     * Whether to delete the file in the event of a failure.
+     */
+    public Optional<Boolean> deleteOnFailure() {
+        return Optional.ofNullable(deleteOnFailure);
+    }
+
+    /**
+     * Create a {@link Builder}, used to create a {@link FileTransformerConfiguration}.
+     */
+    public static Builder builder() {
+        return new DefaultBuilder();
+    }
+
+    @Override
+    public Builder toBuilder() {
+        return new DefaultBuilder().fileWriteOption(fileModificationOption)
+                                   .deleteOnFailure(deleteOnFailure);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        FileTransformerConfiguration that = (FileTransformerConfiguration) o;
+
+        if (fileModificationOption != that.fileModificationOption) {
+            return false;
+        }
+        return Objects.equals(deleteOnFailure, that.deleteOnFailure);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = fileModificationOption != null ? fileModificationOption.hashCode() : 0;
+        result = 31 * result + (deleteOnFailure != null ? deleteOnFailure.hashCode() : 0);
+        return result;
+    }
+
+    public enum FileWriteOption {
+        /**
+         * Always create a new file. If the file already exists, {@link FileAlreadyExistsException} will be thrown.
+         * In the event of an error, the SDK will attempt to delete the file (whatever has been written to it so far).
+         */
+        CREATE_NEW,
+
+        /**
+         * Create a new file if it doesn't exist, otherwise replace the existing file.
+         * <p>
+         * In the event of an error, the SDK will NOT attempt to delete the file.
+         */
+        CREATE_OR_REPLACE_EXISTING,
+
+        /**
+         * Create a new file if it doesn't exist, otherwise append the existing file.
```

Comment:
minor , "append to". Appending the existing file would be very different behavior!

Reply:
Good call!